v-next/hardhat-chai-matchers/src/utils.ts

```diff
@@ -0,0 +1,57 @@
+import { HardhatError } from "@ignored/hardhat-vnext-errors";
+import { AssertionError } from "chai";
+
+// just a generic function type to avoid errors from the ban-types eslint rule
+export type Ssfi = (...args: any[]) => any;
+
+/**
+ * This function is used by the matchers to obtain an `assert` function, which
+ * should be used instead of `this.assert`.
+ *
+ * The first parameter is the value of the `negated` flag. Keep in mind that
+ * this value should be captured at the beginning of the matcher's
+ * implementation, before any async code is executed. Otherwise things like
+ * `.to.emit().and.not.to.emit()` won't work, because by the time the async part
+ * of the first emit is executed, the `.not` (executed synchronously) has already
+ * modified the flag.
+ *
+ * The second parameter is what Chai calls the "start stack function indicator",
+ * a function that is used to build the stack trace. It's unclear to us what's
+ * the best way to use this value, so this needs some trial-and-error. Use the
+ * existing matchers for a reference of something that works well enough.
+ */
+export function buildAssert(negated: boolean, ssfi: Ssfi) {
```

Comment:
Is this function intended for direct use by users? My guess is that it’s not, so I believe it can be moved to the internal folder. However, I need a second opinion to confirm my assumption